src/app/sign-up-metamask/sign-up-metamask.component.ts

```diff
@@ -0,0 +1,255 @@
+import { Component, OnInit } from '@angular/core';
+import { ethers } from 'ethers';
+import {
+  BackendAPIService,
+  TransactionSpendingLimitResponse,
+} from '../backend-api.service';
+import { LoginMethod, Network } from '../../types/identity';
+import { ec } from 'elliptic';
+import { CryptoService } from '../crypto.service';
+import { getSpendingLimitsForMetamask } from '../log-in/log-in.component';
+import { AccountService } from '../account.service';
+import { IdentityService } from '../identity.service';
+import { GlobalVarsService } from '../global-vars.service';
+import { SigningService } from '../signing.service';
+import { MetamaskService } from '../metamask.service';
+enum SCREEN {
+  CREATE_ACCOUNT = 0,
+  LOADING = 1,
+  ACCOUNT_SUCCESS = 2,
+  AUTHORIZE_MESSAGES = 3,
+  MESSAGES_SUCCESS = 4,
+}
+enum METAMASK {
+  START = 0,
+  CONNECT = 1,
+  ERROR = 2,
+}
+@Component({
+  selector: 'app-sign-up-metamask',
+  templateUrl: './sign-up-metamask.component.html',
+  styleUrls: ['./sign-up-metamask.component.scss'],
+})
+export class SignUpMetamaskComponent implements OnInit {
+  private static UNLIMITED_DERIVED_KEY_EXPIRATION: Readonly<number> = 100000000000;
+  metamaskState: METAMASK = METAMASK.START;
+  currentScreen: SCREEN = SCREEN.CREATE_ACCOUNT;
+  timer: any;
+  SCREEN = SCREEN;
+  METAMASK = METAMASK;
+  publicKey = '';
+  errorMessage = '';
+
+  constructor(
+    private accountService: AccountService,
+    private identityService: IdentityService,
+    private cryptoService: CryptoService,
+    public globalVars: GlobalVarsService,
+    private backendApi: BackendAPIService,
+    private signingService: SigningService,
+    private metamaskService: MetamaskService
+  ) {}
+
+  ngOnInit(): void {}
```

Comment:
If we're not using the onInit, please remove the `implements OnInit` from the header and delete this.

Reply:
Yup!